--- a/plugins/input.h
+++ b/plugins/input.h
@@ -20,6 +20,7 @@
 #                                                                              #
 *******************************************************************************/
 
+#include <pthread.h>
 #include <syslog.h>
 #include "../mjpg_streamer.h"
 #define INPUT_PLUGIN_PREFIX " i: "
--- a/plugins/input_uvc/v4l2uvc.h
+++ b/plugins/input_uvc/v4l2uvc.h
@@ -36,6 +36,7 @@
 #include <sys/mman.h>
 #include <sys/select.h>
 #include <sys/time.h>
+#include <pthread.h>
 
 #include <linux/types.h>          /* for videodev2.h */
 #include <linux/videodev2.h>
